mat64/dense.go

```diff
@@ -512,7 +512,13 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 	m.mat.Stride = int(cols)
 	m.capRows = int(rows)
 	m.capCols = int(cols)
-	m.mat.Data = use(m.mat.Data, m.mat.Rows*m.mat.Cols)
+
```

Comment:
They need to individually checked for being negative. 
Move the check before assigning to m. 
I'm not sure that check for overflow is correct, on 64 bit systems the matrix could be larger. The check can be something like 
````
size := rows * cols
if size < 0 || int64(int(size)) != size { ...}
````